snuba/datasets/storages/spans.py

```diff
@@ -0,0 +1,90 @@
+from snuba.clickhouse.columns import (
+    UUID,
+    Array,
+    ColumnSet,
+    DateTime,
+    LowCardinality,
+    Materialized,
+    Nested,
+    String,
+    UInt,
+    WithDefault,
+)
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.datasets.schemas import MandatoryCondition
+from snuba.datasets.schemas.tables import ReplacingMergeTreeSchema
+from snuba.datasets.spans_processor import UNKNOWN_SPAN_STATUS, SpansMessageProcessor
+from snuba.datasets.storage import WritableTableStorage
+from snuba.datasets.storages import StorageKey
+from snuba.datasets.storages.tags_hash_map import TAGS_HASH_MAP_COLUMN
+from snuba.datasets.table_storage import KafkaStreamLoader
+from snuba.query.conditions import ConditionFunctions, binary_condition
+from snuba.query.expressions import Column, Literal
+from snuba.query.processors.prewhere import PrewhereProcessor
+from snuba.web.split import TimeSplitQueryStrategy
+
+columns = ColumnSet(
+    [
+        ("project_id", UInt(64)),
+        ("transaction_id", UUID()),  # event_id of the transaction
+        ("span_id", UInt(64)),
+        ("transaction_name", LowCardinality(String())),
+        ("parent_span_id", UInt(64)),  # This should be able to replace transaction_id
+        ("description", LowCardinality(String())),  # description in span
+        ("op", LowCardinality(String())),
+        ("status", WithDefault(UInt(8), str(UNKNOWN_SPAN_STATUS)),),
+        ("start_ts", DateTime()),
+        ("start_ns", UInt(32)),
+        ("finish_ts", DateTime()),
+        ("finish_ns", UInt(32)),
+        ("duration", UInt(32)),
+        ("tags", Nested([("key", String()), ("value", String())])),
+        ("_tags_hash_map", Materialized(Array(UInt(64)), TAGS_HASH_MAP_COLUMN)),
+        ("retention_days", UInt(16)),
+        ("deleted", UInt(8)),
+    ]
+)
+
+schema = ReplacingMergeTreeSchema(
+    columns=columns,
+    local_table_name="spans_experimental_local",
+    dist_table_name="spans_experimental_dist",
+    storage_set_key=StorageSetKey.TRANSACTIONS,
+    mandatory_conditions=[
+        MandatoryCondition(
+            ("deleted", "=", 0),
+            binary_condition(
+                None,
+                ConditionFunctions.EQ,
+                Column(None, None, "deleted"),
+                Literal(None, 0),
+            ),
+        )
+    ],
+    prewhere_candidates=["transaction_id", "description"],
+    order_by="(project_id, toStartOfDay(finish_ts), cityHash64(parent_span_id), op, cityHash64(span_id))",
```

Comment:
project_id is there as usual.
Then there are two main use cases for this table: accessed via left join from the transactions table (major) or accessed directly by time range without join. 
For the join we would want the parent_span_id, which identifies the transaction, to appear first. For the time range queries we would prefer the finish_ts to appear first.
If we put the parent_span_id first, the join would work very well but the time range queries would be abysmal since we would have to do a full table scan. So we put the timestamp bucketed by day first. Which is good for time range and it is not terrible for joins, since the buckets to look into are few and generally queries are already limited by time range anyway. 
We can try to add a bf/set index to the parent_span_id to improve the join performance.
op is useful to find specific types of spans in long transactions and span_id is needed because this is also the primary key of the table.